restructuredtext_lint/test/test.py

```diff
@@ -129,9 +129,50 @@ def test_rst_lint_many_files(self):
             # python ../cli.py test_files/valid.rst invalid.rst
             subprocess.check_output((sys.executable, rst_lint_path, valid_rst, invalid_rst))
         output = str(e.exception.output)
-        # 'rst-lint' should exit with error code 1:
-        self.assertEqual(e.exception.returncode, 1)
+        # 'rst-lint' should exit with error code 2 as linting failed:
+        self.assertEqual(e.exception.returncode, 2)
         # There should be a least one valid .rst file:
         self.assertIn('is clean', output)
         # There should be a least one invalid rst file:
         self.assertIn('WARNING', output)
+
+    def test_report_levels(self):
+        """A document which triggers linting warnings, to test --level argument"""
+        # Expect to get either two lines (both warnings), or no lines of output
+        for level, count in ((1, 2), (2, 2), (3, 0), (4, 0)):
+            with self.assertRaises(subprocess.CalledProcessError) as e:
+                # Note we want to ensure stdout is a string (not bytes)
+                subprocess.check_output((sys.executable, rst_lint_path, '--level', str(level), warning_rst),
+                                        universal_newlines=True)
+            output = str(e.exception.output)
+            # 'rst-lint' should exit with error code 2 as linting failed:
+            self.assertEqual(e.exception.returncode, 2)
+            self.assertEqual(count, output.count('\n'), output)
+            self.assertEqual(count, output.count('WARNING'), output)
+
+    def test_bad_level(self):
+        """Confirm return code 1 from bad --level argument"""
+        with self.assertRaises(subprocess.CalledProcessError) as e:
+            # Want to hide the error message from the test output
+            subprocess.check_output((sys.executable, rst_lint_path, '--level', '5', warning_rst),
+                                    stderr=subprocess.DEVNULL)
+            # 'rst-lint' should exit with error code 1 as bad argument:
+            self.assertEqual(e.exception.returncode, 1)
```

Comment:
Having been bitten by command line API changes in the past, personally I'd include some in the test suite.